cpp/tests/copying/copy_tests.cu

```diff
@@ -0,0 +1,87 @@
+#include <cudf/column/column_device_view.cuh>
+
+template <typename T>
+struct CopyTest : public cudf::test::BaseFixture {};
+
+TYPED_TEST_CASE(CopyTest, cudf::test::NumericTypes);
+
+// to keep names shorter
+#define wrapper cudf::test::fixed_width_column_wrapper
+using bool_wrapper = wrapper<cudf::experimental::bool8>;
+
+TYPED_TEST(CopyTest, CopyIfElseTest) 
+{ 
+   using T = TypeParam;   
+      
+   // make sure we span at least 2 warps
+   int num_els = 64;
+
+   bool _mask[]   = { 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+                      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 };
+   bool_wrapper mask_w(_mask, _mask + num_els);
+
+   T _lhs[]       = { 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 
+                      5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5 };
+   bool _lhs_v[]  = { 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+                      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 };
+   wrapper<T> lhs_w(_lhs, _lhs + num_els, _lhs_v);      
+
+   T _rhs[]       = { 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6,
+                      6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6 };
+   bool _rhs_v[]  = { 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+                      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 };
+   wrapper<T> rhs_w(_rhs, _rhs + num_els, _rhs_v);            
+
+   T _expected[]  = { 5, 6, 5, 6, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 6, 5, 5, 5, 5, 5, 5, 5, 5, 5, 
+                      5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5 };
+   bool _exp_v[]  = { 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+                      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 };
+   wrapper<T> expected_w(_expected, _expected + num_els, _exp_v);
+
+   // construct input views
+   cudf::column mask(mask_w);
+   cudf::column_view mask_v(mask);
+   //
+   cudf::column lhs(lhs_w);
+   cudf::column_view lhs_v = lhs.view();
+   //
+   cudf::column rhs(rhs_w);
+   cudf::column_view rhs_v = rhs.view();
+   //
+   cudf::column expected(expected_w);
+   cudf::column_view expected_v = expected.view();
+
+   // get the result
+   auto out = cudf::experimental::copy_if_else(lhs_v, rhs_v, mask_v);
+   cudf::column_view out_v = out->view();      
+
+   // compare
+   cudf::test::expect_columns_equal(out_v, expected_v);   
+}
```

Comment:
Should also test corner cases: empty inputs, error cases like type mismatch, etc.